server/src/main/kotlin/tech/figure/objectstore/gateway/configuration/AppConfig.kt

```diff
@@ -23,11 +23,15 @@ class AppConfig {
     }
 
     @Bean(BeanQualifiers.OBJECTSTORE_ENCRYPTION_KEYS)
-    fun encryptionKeys(provenanceProperties: ProvenanceProperties, objectStoreProperties: ObjectStoreProperties): Map<String, KeyRef> = objectStoreProperties.privateKeys.map {
-        it.toJavaPrivateKey().toKeyPair().let { keyPair ->
-            keyPair.public.getAddress(provenanceProperties.mainNet) to DirectKeyRef(keyPair)
-        }
-    }.toMap()
+    fun encryptionKeys(provenanceProperties: ProvenanceProperties, objectStoreProperties: ObjectStoreProperties): Map<String, KeyRef> =
+        objectStoreProperties.privateKeys
+            .filterNot { it.isBlank() }
+            .filterNot { it.isEmpty() }
```

Comment:
the isBlank check takes care of empty as well `Returns true if this string is empty or consists solely of whitespace characters.`